### ISSUE TITLE
array value can't be queried if there's an index on the field

### ISSUE THREAD

**Issue by askoog** [bwaldvogel/mongo-java-server#201]
Given a database index on a field where the values are an array, it is not possible to find the document using an exact document template, you'll need to specify an `$all` option.

Example java code:

```
InMemoryMongoServer server = new InMemoryMongoServer();
server.bind();
MongoClient mongoClient = new MongoClient(new ServerAddress(server.getLocalAddress()));
MongoDatabase db = mongoClient.getDatabase("db");
MongoCollection<Document> collection = db.getCollection("coll");


collection.createIndex(new Document("key", 1), new IndexOptions().unique(true));
collection.insertOne(new Document("key", Arrays.asList("val1", "val2")));

// search for { "key" : [ "val1", "val2" ]}
FindIterable<Document> documents = collection.find(new Document("key", Arrays.asList("val1", "val2")));
List<Document> collect = StreamSupport.stream(documents.spliterator(), false).collect(Collectors.toList());
// no entries found
System.out.println(collect);

// search for { "key" : { "$all" : [ "val1", "val2" ]}}
documents = collection.find(new Document("key", new Document("$all", Arrays.asList("val1", "val2"))));
collect = StreamSupport.stream(documents.spliterator(), false).collect(Collectors.toList());
// finds the document
System.out.println(collect);

```

According to the mongo [documentation](https://www.mongodb.com/docs/v5.0/tutorial/query-arrays/#match-an-array), I'd expect the first query to also return the document, which is also the case if I skip the index creation.

A long shot guess is that the code handling [index updates](https://github.com/bwaldvogel/mongo-java-server/blob/main/core/src/main/java/de/bwaldvogel/mongo/backend/AbstractUniqueIndex.java#L94) maybe also should create an entry for the array data as well?

Thanks for a great tool by the way!

**Comment by bwaldvogel** [comment]
Thanks for reporting this issue. It should be fixed with 36a4bb3e139d8e444c6b2f72674c80c9d725de08. You could re-test with the latest version on `main`?

**Comment by bwaldvogel** [comment]
The fix is available in version `1.42.0`.
